factory/usr/lib/systemd/system/initrd-root-fs.target.d/42-kernel.conf

```diff
@@ -1,5 +1,4 @@
 [Unit]
-Requires=run-mnt-kernel.mount
```

Comment:
just to be clear, the Requires here are being removed because they are superfluous as you mention in your other comment?

Reply:
Those do not exist at boot. They are generated by `the-tool.service`. In practice I have noticed it tends to confuse systemd because the graph of jobs is built at load. So generating new units requires `daemon-reload`. Instead we can just make sure to have `Wants=the-tool.service` and `After=the-tool.service` and remove the mentions to `run-mnt-*.mount`. We only need to keep them in the `Wants` of the initrd targets.

Reply:
Thanks for confirming